xarray/backends/api.py

```diff
@@ -550,11 +557,15 @@ def maybe_decode_store(store, chunks, lock=False):
             extra_kwargs["lock"] = lock
         elif engine == "zarr":
             backend_kwargs = backend_kwargs.copy()
+            backend_kwargs.pop("fs", None)
             overwrite_encoded_chunks = backend_kwargs.pop(
                 "overwrite_encoded_chunks", None
             )
             extra_kwargs["mode"] = "r"
             extra_kwargs["group"] = group
+            if fs is not None:
+                filename_or_obj = fs.get_mapper(filename_or_obj)
```

Comment:
Note that there we are working on refactor of the backend API that among other things aims at removing all knowledge of what backends can or can't do from `open_dataset`. Adding logic inside `if engine == "zarr"` will probably result in merge conflicts.

I would suggest to move the call to `fs.get_mapper(filename_or_obj)` inside the zarr backend.

Reply:
Thanks for the heads up. I already did one slightly complex conflict resolve.

It isn't totally clear, though, that the logic can be buried in the zarr engine for two reasons:
- when using open_mf, the globbing of remote files/directories happens early, before establishing individual zarr instances
- actually the file instances that fsspec makes from URLs can be used by some other backends; that just happens not the be the emphasis here

Happy to go whichever way is most convenient for the library.

Reply:
We need to resolve this discussion in order to decide what to do about this PR. Any more thoughts from other devs.

In my view, some of the fsspec logic introduced in the PR should eventually move to the generic `open_mfdataset` function, as it is not really specific to Zarr. However, I don't see a strong downside to adding it to `open_zarr` right now. Eventually `open_zarr` will be deprecated. But the pattern used here could be copied and incorporated into the backend refactor.